README.md

```diff
@@ -258,6 +263,43 @@ final ExecutorService zosmaExecutor = Executors.newSingleThreadExecutor();
 final Future futureZosma = zosmaExecutor.submit(chessZosma); 
 ```
 
+### Suppliers
+
+Since the data evolves as it is processed the type of the data container may change as well. In 
+order to allow composition of these data containers through `Deducer`s, these `Deducer`s should 
+only rely interfaces instead of class types. Unfortunately Java's `Supplier` 
```

Comment:
'rely on' or relay?

Reply:
rely on would fit better. but what about `should be parametrized by`

Reply:
Sounds more concrete, yeah